pkg/controllers/nodeclass/status/ami_test.go

```diff
@@ -89,133 +91,426 @@ var _ = Describe("NodeClass AMI Status Controller", func() {
+			ExpectObjectReconciled(ctx, env.Client, statusController, nodeClass)
+			nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+
+			Expect(len(nodeClass.Status.AMIs)).To(Equal(6))
+			Expect(nodeClass.Status.AMIs).To(ContainElements([]v1.AMI{
+				{
+					Name: "amd64-standard",
+					ID:   "ami-amd64-standard",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureAmd64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
+						{
+							Key:      v1.LabelInstanceAcceleratorCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
 					},
 				},
-			},
-			{
-				Name: "test-ami-2",
-				ID:   "ami-id-456",
-				Requirements: []corev1.NodeSelectorRequirement{
-					{
-						Key:      corev1.LabelArchStable,
-						Operator: corev1.NodeSelectorOpIn,
-						Values:   []string{karpv1.ArchitectureAmd64},
+				{
+					Name: "arm64-standard",
+					ID:   "ami-arm64-standard",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureArm64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
+						{
+							Key:      v1.LabelInstanceAcceleratorCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
 					},
-					{
-						Key:      v1.LabelInstanceAcceleratorCount,
-						Operator: corev1.NodeSelectorOpExists,
+				},
+				{
+					Name: "amd64-nvidia",
+					ID:   "ami-amd64-nvidia",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureAmd64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
+							Operator: corev1.NodeSelectorOpExists,
+						},
 					},
 				},
-			},
-			{
-				Name: "test-ami-1",
-				ID:   "ami-id-123",
-				Requirements: []corev1.NodeSelectorRequirement{
-					{
-						Key:      corev1.LabelArchStable,
-						Operator: corev1.NodeSelectorOpIn,
-						Values:   []string{karpv1.ArchitectureAmd64},
+				{
+					Name: "arm64-nvidia",
+					ID:   "ami-arm64-nvidia",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureArm64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
+							Operator: corev1.NodeSelectorOpExists,
+						},
 					},
-					{
-						Key:      v1.LabelInstanceGPUCount,
-						Operator: corev1.NodeSelectorOpDoesNotExist,
+				},
+				// Note: Bottlerocket uses the same AMI for nvidia and neuron, we use the nvidia AMI here
```

Comment:
so this was only an issue with al2023, yes?

Reply:
looks like this note has a typo `Bottlerocket` -> `AL2`

Reply:
The only accelerated AMI vended by Bottlerocket is the nvidia variant so this wasn't a typo. We've historically selected the nvidia AMI for both nvidia and inferentium / trainium instances but looking at this [issue](https://github.com/bottlerocket-os/bottlerocket/issues/1995) I'm wondering if that was incorrect.

But yes, the only AMI family we currently support that will have separate AMIs for gpu and accelerator workloads is AL2023.

Reply:
> We've historically selected the nvidia AMI for both nvidia and inferentium / trainium instances but looking at this https://github.com/bottlerocket-os/bottlerocket/issues/1995 I'm wondering if that was incorrect.

Thats correct - only the EKS AL2 GPU AMI can be used on instances with NVIDIA GPUs and AWS Neuron devices. The Bottlerocket NVIDIA variant will only work on NVIDIA GPUs; it won't work on AWS Neuron devices (though that variant is in the works)

Reply:
> do you know if this AMI is strictly incompatible? 

Yes - you can only use the Bottlerocket NVIDIA variant on instances with NVIDIA GPUs and vice versa with Neuron devices *if you want to use those GPUs/accelerators. And no, nothing can fix this in user data - the NVIDIA driver and Neuron driver are baked into the OS which is immutable once created

> Either way, I think we can take that as a follow-up rather than changing that here.

Yes - I will defer to you all in terms of how to best tackle this

Reply:
Discussed with the team today, we'll plan on making this change at v1.1.0. While it shouldn't affect any users, and it is addressing a bug, it's still a breaking change so I'd like to all it out as such. Since we're going to be backporting this PR to 1.0, we'll track that change in a separate PR.